repo2docker/buildpacks/base.py

```diff
@@ -286,25 +278,24 @@ def compose_with(self, other):
         for resolving them.
         """
         result = BuildPack(parent=self)
-        labels = {}
-        labels.update(self.labels)
-        labels.update(other.labels)
-        result.labels = labels
-        result.packages = self.packages.union(other.packages)
-        result.base_packages = self.base_packages.union(other.base_packages)
-        result.path = self.path + other.path
-        # FIXME: Deduplicate Env
-        result.env = self.env + other.env
-        result.build_scripts = self.build_scripts + other.build_scripts
-        result.assemble_scripts = (self.assemble_scripts +
-                                   other.assemble_scripts)
-        result.post_build_scripts = (self.post_build_scripts +
-                                     other.post_build_scripts)
-
-        build_script_files = {}
-        build_script_files.update(self.build_script_files)
-        build_script_files.update(other.build_script_files)
-        result.build_script_files = build_script_files
+        # FIXME: Temporary hack so we can refactor this piece by piece instead of all at once!
+
+        def _merge_dicts(d1, d2):
+            md = {}
+            md.update(d1)
+            md.update(d2)
+            return md
+
+        result.get_packages = lambda: self.get_packages().union(other.get_packages())
```

Comment:
Do you think it's appropriate to define a `ComposedBuildPacks` rather than redefining it at runtime with lambdas here?

Reply:
Nah, I think we should just wait and move directly to inheritance in another PR (pretty soon after this)

Reply:
ok